### PR TITLE
Add optional resource parameters to census resource test

### PR DESCRIPTION
**Pull request by kpayson64**
Needed for running in Google3.

cc @a-veitch

### CODE REVIEW

File: test/core/census/resource_test.c [hunk 1]
```diff
@@ -136,22 +133,34 @@ static void test_base_resources() {
 }
 
 int main(int argc, char **argv) {
+  char *resource_empty_name_pb, *resource_full_pb, *resource_minimal_good_pb,
+  *resource_no_name_pb, *resource_no_numerator_pb, *resource_no_unit_pb;
+  if (argc >= 7) {
```
Comment (a-veitch) [inline test/core/census/resource_test.c]: I think better would be something like:

if (argc == 7) {
   set params from args...;
} else if (argc == 1) {
   set params to explicit defaults;
} else {
  fail test with message about wrong # of args;
}

File: test/core/census/resource_test.c [hunk 2]
```diff
@@ -136,22 +133,34 @@ static void test_base_resources() {
 }
 
 int main(int argc, char **argv) {
+  char *resource_empty_name_pb, *resource_full_pb, *resource_minimal_good_pb,
```
Comment (a-veitch) [inline test/core/census/resource_test.c]: make these const?

File: test/core/census/resource_test.c [hunk 3]
```diff
@@ -95,15 +95,13 @@ static void test_define_single_resource(const char *file, const char *name,
 }
 
 // Try deleting various resources (both those that exist and those that don't).
-static void test_delete_resource() {
+static void test_delete_resource(char* minimal_good, char* full) {
```
Comment (a-veitch) [inline test/core/census/resource_test.c]: make arg types 'const char*'

**Comment by kpayson64** [comment]
Addressed feedback.

**Comment by dgquintas** [comment]
Sanity is failing, clang-format https://grpc-testing.appspot.com/job/gRPC_pull_requests/11982/config=opt,language=sanity,platform=linux/console

**Comment by kpayson64** [comment]
Relevant tests are now passing.

**Comment by dgquintas** [comment]
LGTM

**Pull request merged**